dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -113,17 +113,14 @@ public Host resolveHostName(String serverName, User user, boolean respectFronten
         User systemUser = APILocator.systemUser();
 
         if(host == null){
+
             try {
-                host = resolveHostNameWithoutDefault(serverName, systemUser, respectFrontendRoles).get();
+                final Optional<Host> optional = resolveHostNameWithoutDefault(serverName, systemUser, respectFrontendRoles);
+                host = optional.isPresent() ? optional.get() : findDefaultHost(systemUser, respectFrontendRoles);
             } catch (Exception e) {
-                return findDefaultHost(systemUser, respectFrontendRoles);
-            }
-            
-            //If no host matches then we set the default host.
-            if(host == null){
                 host = findDefaultHost(systemUser, respectFrontendRoles);
```

Comment:
Also log something for support

Reply:
I think this is normal behavior here - if we do log something, log a debug with the supplier interface